packages/blade/src/components/Tag/Tag.stories.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import type { DOMAttributes } from 'react';
+import type { Meta, ComponentStory } from '@storybook/react';
+import { Title } from '@storybook/addon-docs';
+import type { TagProps } from './Tag';
+import { Tag } from './Tag';
+import { Sandbox } from '~utils/storybook/Sandbox';
+import StoryPageWrapper from '~utils/storybook/StoryPageWrapper';
+import { getStyledPropsArgTypes } from '~components/Box/BaseBox/storybookArgTypes';
+import { Box } from '~components/Box';
+import { Button } from '~components/Button';
+import { TextInput } from '~components/Input/TextInput';
+import { isReactNative } from '~utils';
+import { PlusIcon } from '~components/Icons';
+import iconMap from '~components/Icons/iconMap';
+
+const Page = (): React.ReactElement => {
+  return (
+    <StoryPageWrapper
+      figmaURL={{
+        paymentTheme:
+          'https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=28435%3A581488',
+        bankingTheme:
+          'https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=15234%3A480939',
+      }}
+      componentName="Tag"
+      componentDescription="These are set of interactive keywords that help organise & categorise objects. Tags can be added or removed from an object by the users."
+    >
+      <Title>Usage</Title>
+      <Sandbox>
+        {`
+        import React from 'react';
+        import { Tag, FileTextIcon } from '@razorpay/blade/components';
+        
+        function App(): JSX.Element {
+          const [isTagVisible, setIsTagVisible] = React.useState(true);
+
+          return (
+            isTagVisible 
+            ? <Tag
+                icon={FileTextIcon}
+                onDismiss={() => {
+                  console.log('Unpaid Tag dismissed');
+                  setIsTagVisible(false);
+                }}
+              >
+                Unpaid
+              </Tag>
+            : null
+          )
+        }
+
+        export default App;
+        `}
+      </Sandbox>
+    </StoryPageWrapper>
+  );
+};
+
+export default {
+  title: 'Components/Tag',
+  component: Tag,
+  argTypes: {
+    ...getStyledPropsArgTypes(),
+    icon: {
+      name: 'icon',
+      type: 'select',
+      options: Object.keys(iconMap),
+      mapping: iconMap,
+    },
+  },
+  parameters: {
+    docs: {
+      page: Page,
+    },
+  },
+} as Meta<TagProps>;
+
+const TagTemplate: ComponentStory<typeof Tag> = ({ children, ...args }) => {
+  const [isTagVisible, setIsTagVisible] = React.useState(true);
+  return (
+    <Box>
+      {isTagVisible ? (
+        <Tag {...args} onDismiss={() => setIsTagVisible(false)}>
+          {children}
+        </Tag>
+      ) : null}
+    </Box>
+  );
+};
+
+export const Default = TagTemplate.bind({});
+Default.args = {
+  children: 'Unpaid',
+  icon: 'FileTextIcon',
+} as TagProps & { icon: string };
+
+export const Disabled = TagTemplate.bind({});
```

Comment:
Will have to change the focus ring on our IconButton components. Mentioned it to design here - https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps?node-id=29744:568562&mode=dev#509768673

Reply:
so do we release it as it is?

Reply:
I guess yes. Don't see any way to solve this without touching IconButton's focus ring. We were also discussing about increasing focus area of icon button so that will also change the focus ring styles by default.